packages/interpretations/src/components/html-editor/RichEditor.js

```diff
@@ -0,0 +1,255 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from 'material-ui/Dialog';
+import { Portal } from 'react-portal';
+import { findIndex } from 'lodash/fp';
+import i18n from '@dhis2/d2-i18n'
+import CKEditor from './CKEditor';
+
+const styles = {
+    mentions: {
+        position: "absolute",
+        boxShadow: "rgb(136, 136, 136) 0px 0px 4px 0px",
+        margin: 0,
+        padding: 4,
+        zIndex: 1000000,
+        backgroundColor: "#FAFFFA",
+        listStyleType: "none",
+        fontFamily: "Roboto, sans-serif",
+        maxHeight: "200px",
+        overflowY: "auto"
+    },
+    mentionTitle: {
+        padding: "5px 15px",
+        borderBottom: "1px solid rgb(224, 224, 224)",
+        fontSize: "0.75rem",
+        fontWeight: 500
+    },
+    userMention: {
+        cursor: "pointer",
+        padding: "5px 15px",
+        borderBottom: "1px solid rgb(224, 224, 224)",
+        fontSize: "0.75rem"
+    },
+    userMentionSelected: {
+        backgroundColor: "#ACD",
+    },
+};
+
+class UserMatch extends Component {
```

Comment:
Sorry, I just noticed that you have implemented a component for giving suggestions when a user mentions another user.

@edoardo is working on a wrapper component in https://github.com/dhis2/d2-ui/tree/task/mentions-wrapper which we can use to wrap a text field or text area, e.g.

```
<MentionsWrapper d2={d2}>
    <TextField
        multiline
        rowsMax="4"
        placeholder="Type some text. @ triggers the mentions suggestions"
    />
</MentionsWrapper>
```

This is to get a pluggable suggestion component we can reuse instead of implementing it in each application. 

Going forward we are looking into deprecating CkEditor completely as it is a rather large dependency (500+ KB), and we want to let each application be standalone and not depend on e.g. core-resources.

I tried adding the `MentionsWrapper` to the interpretations panel in master, and it mostly works. There are a couple of bugs which @edoardo is working on and the components needs more testing before it's ready to merge, but the plan is to get it in for 2.30.

The requirements here are as far as I understand it:

1. Mentioning users
2. Suggestion box for usernames
3. Emoticons
4. Rich-text editor

Using the `MentionsWrapper` solves 1. and 2., which add a lot of value, so I think that would be a good start. I don't think it's reasonable nor possible to combine the `MentionsWrapper` with the CkEditor however, so I would rather we just use `TextField`/`textarea` + `MentionsWrapper` for this iteration until we figure out what to do with requirements 3 and 4.